File: julia/2022/day_01.jl
function calorie_count(str::AbstractString)::UInt
    sum(parse(UInt, s) for s in eachsplit(str))
end

function main()
    text = read("input/2022/input_01.txt", String)
    counts = [calorie_count(s) for s in eachsplit(text, "\n\n")]
    partialsort!(counts, 3, rev=true)
    println("Part 1: $(counts[1])")
    println("Part 2: $(sum(counts[1:3]))")
end

main()

File: rust/2022/src/bin/day_01.rs
fn main() {
    let input = aoc::read_input("2022/input_01.txt");
    let mut counts: Vec<u32> = input
        .split("\n\n")
        .map(|block| block.lines().map(|s| s.parse::<u32>().unwrap()).sum())
        .collect();
    counts.sort_by_key(|&x| std::cmp::Reverse(x));
    println!("Part 1: {}", counts[0]);
    println!("Part 2: {}", counts[..3].iter().sum::<u32>());
}
